### PR TITLE
add aws endpoint to bucket notifications

### PR DESCRIPTION
**Pull request by yuvalif**
---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_main.cc [hunk 1]
```diff
@@ -401,6 +402,11 @@ int radosgw_Main(int argc, const char **argv)
     if (!rgw::kafka::init(cct.get())) {
         dout(1) << "ERROR: failed to initialize Kafka manager" << dendl;
     }
+#endif
+#ifdef WITH_RADOSGW_AWS_ENDPOINT
+  if (!rgw::aws::init(cct.get())) {
```
Comment (cbodley) [inline src/rgw/rgw_main.cc:407]: we're adding global init/shutdown to rgw_main.cc - do we need it in rgw_admin.cc or librgw.cc?

**Review by cbodley — COMMENTED** [review]
can we please rename rgw_aws.* to rgw_notify_aws.* so its a bit more specific?

File: src/rgw/rgw_main.cc (hunk 1 above)
Comment (yuvalif) [reply src/rgw/rgw_main.cc:407]: not needed in rgw_admin
*is* needed in librgw (to avoid the persistent notification "ghost rgw" bug...) - will fix!

**Comment by yuvalif** [comment]
> can we please rename rgw_aws.* to rgw_notify_aws.* so its a bit more specific?

sure, but note  thatwe already have "rgw_amqp*" and "rgw_kafka*". should we change there as well, or is there meaning there clearer?

**Comment by cbodley** [comment]
> > can we please rename rgw_aws.* to rgw_notify_aws.* so its a bit more specific?
> 
> sure, but note thatwe already have "rgw_amqp*" and "rgw_kafka*". should we change there as well, or is there meaning there clearer?

i do think that would make it easier for people unfamiliar with the codebase to find all the bits related to notifications, yeah

**Comment by yuvalif** [comment]
> > > can we please rename rgw_aws.* to rgw_notify_aws.* so its a bit more specific?
> > 
> > 
> > sure, but note thatwe already have "rgw_amqp*" and "rgw_kafka*". should we change there as well, or is there meaning there clearer?
> 
> i do think that would make it easier for people unfamiliar with the codebase to find all the bits related to notifications, yeah

ok. will split to a different commit

**Review by cbodley — APPROVED** [review]

**Comment by yuvalif** [comment]
@cbodley @mattbenjamin could you please approve the modification to the submodule?

**Comment by yuvalif** [comment]
shaman build in ubuntu are silently failing :-(

File: CMakeLists.txt [hunk 2]
```diff
@@ -409,6 +409,7 @@ option(WITH_RADOSGW_BEAST_OPENSSL "Rados Gateway's Beast frontend uses OpenSSL"
 option(WITH_RADOSGW_AMQP_ENDPOINT "Rados Gateway's pubsub support for AMQP push endpoint" ON)
 option(WITH_RADOSGW_KAFKA_ENDPOINT "Rados Gateway's pubsub support for Kafka push endpoint" ON)
 option(WITH_RADOSGW_LUA_PACKAGES "Rados Gateway's support for dynamically adding lua packagess" ON)
+option(WITH_RADOSGW_AWS_ENDPOINT "Rados Gateway's pubsub support for AWS push endpoint" ON)
```
Comment (anthonyeleven) [inline CMakeLists.txt:412]: s/packagess/packages/
s/lua/Lua/
4x s/Gateway’s/Gateway/

File: doc/radosgw/notifications.rst [hunk 3]
```diff
@@ -7,7 +7,7 @@ Bucket Notifications
 .. contents::
 
 Bucket notifications provide a mechanism for sending information out of the radosgw when certain events are happening on the bucket.
-Currently, notifications could be sent to: HTTP, AMQP0.9.1 and Kafka endpoints.
+Currently, notifications could be sent to: HTTP, AMQP0.9.1, Kafka, AWS lambda and AWS SNS endpoints.
```
Comment (anthonyeleven) [inline doc/radosgw/notifications.rst:10]: s/could/may/

Comment (anthonyeleven) [reply doc/radosgw/notifications.rst:10]: s/://

**Review by anthonyeleven — CHANGES_REQUESTED** [review]

**Comment by mattbenjamin** [comment]
@yuvalif what is blocking this PR?

**Comment by yuvalif** [comment]
> @yuvalif what is blocking this PR?

packaging of the AWS C++ SDK

**Pull request closed without merging**